code/modules/vending/vending.dm

```diff
@@ -1587,6 +1618,385 @@
+			setFrameState("WIRESREMOVED", user)
+		else if (isweldingtool(target) && !wrenched)
+			var/obj/item/weldingtool/T = target
+			if (T.try_weld(user,0,-1,0,1))
+				SETUP_GENERIC_ACTIONBAR(user, src, 2 SECONDS, /obj/machinery/vendingframe/proc/setFrameState,\
+				list("DECONSTRUCTED", user, target), target.icon, target.icon_state, null)
+/obj/machinery/vending/player
+	name = "YouVend"
+	icon_state = "player"
+	desc = "Sells your stuff!"
+	pay = 1
+	layer = OBJ_LAYER - 0.3
+	//Product loading chute
+	var/loading = 0
+	var/unlocked = 0
+	//Registered owner
+	var/owner = null
+	//card display name
+	var/cardname
+	//Bank account
+	var/datum/data/record/owneraccount = null
+	var/image/crtoverlay = null
+	player_list = list()
+
+	New()
+		. = ..()
+		crtoverlay = SafeGetOverlayImage("screen", src.icon, "player-crt")
+		crtoverlay.layer = src.layer + 0.2
+		crtoverlay.plane = PLANE_DEFAULT
+		//These stop the overlay from being selected instead of the item by your mouse?
+		crtoverlay.appearance_flags = NO_CLIENT_COLOR
+		crtoverlay.mouse_opacity = 0
+		setCrtOverlayStatus(1)
+
+	proc/pick_product_name()
+		var/datum/data/vending_product/player_product/R = pick(src.player_list)
+		var/itemPromo = sanitize(html_encode(R.product_name))
+		return itemPromo
+
+	proc/generate_slogans()
+		if(!length(player_list) <= 0)
+			slogan_list = list("By popular demand: [pick_product_name()]!",
+		"[src.name]. The better vending machine.",
+		"Potentially well stocked!",
+		"Buy my stuff!",
+		"Don't miss out on [pick_product_name()]!",
+		"[src.name]. What else were you going to buy?",
+		"New and improved [pick_product_name()]!")
+
+	proc/getScaledIcon(obj/item/target)
+		var/image/itemoverlayoriginal = null
+		itemoverlayoriginal = SafeGetOverlayImage(null, target, target.icon_state)
+		itemoverlayoriginal.transform = matrix(null, 0.45, 0.45, MATRIX_SCALE)
+		itemoverlayoriginal.pixel_x = -3
+		itemoverlayoriginal.pixel_y = -4
+		itemoverlayoriginal.layer = src.layer + 0.1
+		itemoverlayoriginal.plane = PLANE_DEFAULT
+		return itemoverlayoriginal
+	proc/setItemOverlay(image/target)
+		//Offsets go weird if I don't clear
+		//Also it just breaks randomly if I leave caching on
+		ClearSpecificOverlays(0, "item")
+		UpdateOverlays(target, "item", 0, 1)
+
+	proc/setCrtOverlayStatus(status)
+		if(status == 1)
+			UpdateOverlays(crtoverlay, "screen", 0, 1)
+		else
+			UpdateOverlays(null, "screen", 0, 1)
+
+	proc/addProduct(obj/item/target, mob/user)
+		var/obj/item/storage/targetContainer = target
+		if(!istype(targetContainer))
+			productListUpdater(target, user)
+			user.visible_message("<b>[user.name]</b> loads [target] into [src].")
+			return
+		var/action = input(user, "What do you want to do with [targetContainer]?") as null|anything in list("Empty it into the vending machine","Place it in the vending machine")
+		if(action == "Place it in the vending machine" && !(user.stat || user.restrained() || !in_interact_range(src, user)))
```

Comment:
don't use `mob.stat` directly, macros please
also, check for incapacitation not just restrained, there is a macro for this